hbase-server/src/main/java/org/apache/hadoop/hbase/master/balancer/BaseLoadBalancer.java

```diff
@@ -834,22 +908,20 @@ void regionMoved(int region, int oldServer, int newServer) {
       int tableIndex = regionIndexToTableIndex[region];
       if (oldServer >= 0) {
         numRegionsPerServerPerTable[oldServer][tableIndex]--;
+        // update regionSkewPerTable for the move from old server
+        regionSkewByTable +=
+            Math.abs(numRegionsPerServerPerTable[oldServer][tableIndex]
+              - meanRegionsPerTable[tableIndex])
+              - Math.abs(numRegionsPerServerPerTable[oldServer][tableIndex] + 1
+              - meanRegionsPerTable[tableIndex]);
       }
       numRegionsPerServerPerTable[newServer][tableIndex]++;
-
-      //check whether this caused maxRegionsPerTable in the new Server to be updated
-      if (numRegionsPerServerPerTable[newServer][tableIndex] > numMaxRegionsPerTable[tableIndex]) {
-        numMaxRegionsPerTable[tableIndex] = numRegionsPerServerPerTable[newServer][tableIndex];
-      } else if (oldServer >= 0 && (numRegionsPerServerPerTable[oldServer][tableIndex] + 1)
-          == numMaxRegionsPerTable[tableIndex]) {
-        //recompute maxRegionsPerTable since the previous value was coming from the old server
-        numMaxRegionsPerTable[tableIndex] = 0;
-        for (int[] aNumRegionsPerServerPerTable : numRegionsPerServerPerTable) {
-          if (aNumRegionsPerServerPerTable[tableIndex] > numMaxRegionsPerTable[tableIndex]) {
-            numMaxRegionsPerTable[tableIndex] = aNumRegionsPerServerPerTable[tableIndex];
-          }
-        }
-      }
+      // update regionSkewPerTable for the move to new server
+      regionSkewByTable +=
+        Math.abs(numRegionsPerServerPerTable[newServer][tableIndex]
+          - meanRegionsPerTable[tableIndex])
+          - Math.abs(numRegionsPerServerPerTable[newServer][tableIndex] - 1
+          - meanRegionsPerTable[tableIndex]);
```

Comment:
Is this duplicated code? Can it be made a function?

Reply:
It I s not a duplicate. The previous was "- Math.abs(numRegionsPerServerPerTable[newServer][tableIndex] + 1" vs line 923